rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from json import JSONDecodeError
+from typing import TYPE_CHECKING
+
+import requests
+
+from rotkehlchen.assets.asset import UnderlyingToken
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.ethereum.types import string_to_evm_address
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import YEARN_VAULTS_V1, YEARN_VAULTS_V2_PROTOCOL, ChainID, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+YEARN_OLD_API = 'https://api.yearn.finance/v1/chains/1/vaults/all'
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_yearn_vaults(db: 'DBHandler') -> None:
+    """Query yearn API and ensure that all the tokens exists locally. If it exists but the protocol
+    is not the correct one the asset will be edited.
+    TODO: After Optimism gets merged we need to move this logic to a periodic task as we do with
+    curve pools.
+
+    May raise:
+    - RemoteError
+    """
+    try:
+        response = requests.get(YEARN_OLD_API)
+        data = response.json()
+    except requests.exceptions.RequestException as e:
+        raise RemoteError(f'Failed to obtain yearn vault information. {str(e)}') from e
+    except (DeserializationError, JSONDecodeError) as e:
+        raise RemoteError(f'Failed to deserialize data from cryptoscamdb. {str(e)}') from e
+
+    if not isinstance(data, list):
+        raise RemoteError(f'Unexpected format from yearn vaults reponse. Expected list. {data}')
+
+    for vault in data:
+        if 'v1' in vault['type']:
```

Comment:
From the api docs https://docs.yearn.finance/vaults/yearn-api
> High level vault classification. Valid options are v1 and v2
```suggestion
        if 'v1' == vault['type']:
```

Reply:
Can take this values `APY type. Options are v2:simple, v2:averaged, v1:simple, crv` and also there is `v1`, with nothing else, that is why I use the `in` operator here

Reply:
Ermmmm. you're doing `vault['type']` the only possible values are `v1` and `v2`
Sample response
```json
[
        {
        "inception": 10830536,
        "address": "0xbD17B1ce622d73bD438b9E658acA5996dc394b0d",
        "symbol": "pSLP",
        "name": "pickling SushiSwap LP Token",
        "display_name": "yveCRV - ETH",
        "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0xbD17B1ce622d73bD438b9E658acA5996dc394b0d/logo-128.png",
        "token": {
            "name": "pickling SushiSwap LP Token",
            "symbol": "pSLP",
            "address": "0x5Eff6d166D66BacBC1BF52E2C54dD391AE6b1f48",
            "decimals": 18,
            "display_name": "yveCRV - ETH",
            "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0x5Eff6d166D66BacBC1BF52E2C54dD391AE6b1f48/logo-128.png"
        },
        "tvl": {
            "total_assets": 0,
            "price": 0,
            "tvl": 323530.507
        },
        "apy": {
            "type": "yvecrv-jar",
            "gross_apr": 0.043557184952216506,
            "net_apy": 0.043557184952216506,
            "fees": {
                "performance": null,
                "withdrawal": null,
                "management": null,
                "keep_crv": null,
                "cvx_keep_crv": null
            },
            "points": {
                "week_ago": 0.043557184952216506,
                "month_ago": 0.043557184952216506,
                "inception": 0.043557184952216506
            },
            "composite": null
        },
        "strategies": [],
        "endorsed": true,
        "version": "0.1",
        "decimals": 18,
        "type": "v1",
        "emergency_shutdown": false,
        "updated": 1668155695,
        "migration": null,
        "special": true
    }
]
```
Do you get me now?

Reply:
There is this one 

```
{
      "inception": 10532708,
      "address": "0x597aD1e0c13Bfe8025993D9e79C69E1c0233522e",
      "symbol": "yUSDC",
      "name": "USDC",
      "display_name": "USDC",
      "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0x597aD1e0c13Bfe8025993D9e79C69E1c0233522e/logo-128.png",
      "token": {
         "name": "USD Coin",
         "symbol": "USDC",
         "address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
         "decimals": 6,
         "display_name": "USDC",
         "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/logo-128.png"
      },
      "tvl": {
         "total_assets": 736424464571,
         "price": 0.99997469,
         "tvl": 736405.8256678018
      },
      "apy": {
         "type": "v1:simple",
```

also 

```
   {
      "inception": 11616780,
      "address": "0xe11ba472F74869176652C35D30dB89854b5ae84D",
      "symbol": "yvHEGIC",
      "name": "yvHEGIC 0.3.0",
      "display_name": "HEGIC",
      "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0xe11ba472F74869176652C35D30dB89854b5ae84D/logo-128.png",
      "token": {
         "name": "Hegic",
         "symbol": "HEGIC",
         "address": "0x584bC13c7D411c00c01A62e8019472dE68768430",
         "decimals": 18,
         "display_name": "HEGIC",
         "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0x584bC13c7D411c00c01A62e8019472dE68768430/logo-128.png"
      },
      "tvl": {
         "total_assets": 392711053241553614247328,
         "price": 0.0135279325324709,
         "tvl": 5312.568633007325
      },
      "apy": {
         "type": "v2:simple",
```

Reply:
You're doing `vault['type']` not `vault['apy']['type']`. There's an actual key-value pair for `type`
```json
[
        {
        "inception": 10830536,
        "address": "0xbD17B1ce622d73bD438b9E658acA5996dc394b0d",
        "symbol": "pSLP",
        "name": "pickling SushiSwap LP Token",
        "display_name": "yveCRV - ETH",
        "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0xbD17B1ce622d73bD438b9E658acA5996dc394b0d/logo-128.png",
        "token": {
            "name": "pickling SushiSwap LP Token",
            "symbol": "pSLP",
            "address": "0x5Eff6d166D66BacBC1BF52E2C54dD391AE6b1f48",
            "decimals": 18,
            "display_name": "yveCRV - ETH",
            "icon": "https://rawcdn.githack.com/yearn/yearn-assets/5e9b810dab6ce0a171228f4d0cb37d0d17308e35/icons/multichain-tokens/1/0x5Eff6d166D66BacBC1BF52E2C54dD391AE6b1f48/logo-128.png"
        },
        "tvl": {
            "total_assets": 0,
            "price": 0,
            "tvl": 323530.507
        },
        "apy": {
            "type": "yvecrv-jar",
            "gross_apr": 0.043557184952216506,
            "net_apy": 0.043557184952216506,
            "fees": {
                "performance": null,
                "withdrawal": null,
                "management": null,
                "keep_crv": null,
                "cvx_keep_crv": null
            },
            "points": {
                "week_ago": 0.043557184952216506,
                "month_ago": 0.043557184952216506,
                "inception": 0.043557184952216506
            },
            "composite": null
        },
        "strategies": [],
        "endorsed": true,
        "version": "0.1",
        "decimals": 18,
        "type": "v1",    <--------- here
        "emergency_shutdown": false,
        "updated": 1668155695,
        "migration": null,
        "special": true
    }
]
```